pkg/grafana/dashboard.go

```diff
@@ -0,0 +1,127 @@
+package grafana
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	grafanaOrganizationLabel = "observability.giantswarm.io/organization"
+)
+
+func (s *Service) ConfigureDashboard(ctx context.Context, dashboardCM *v1.ConfigMap) error {
+	return s.processDashboards(ctx, dashboardCM, func(ctx context.Context, dashboard map[string]any, dashboardUID string) {
+		logger := log.FromContext(ctx)
+
+		// Create or update dashboard
+		err := s.PublishDashboard(dashboard)
+		if err != nil {
+			logger.Error(err, "Failed updating dashboard")
+			return
+		}
+
+		logger.Info("updated dashboard", "Dashboard UID", dashboardUID)
```

Comment:
Can you add the uid in the context so it's logge also for the error log?

Reply:
Added the dashboard uid in the context